channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -348,21 +360,30 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControls {
                     let customInterface = senderConfig.httpXHROverride;
                     let httpInterface: IXHROverride = null;
                     let syncInterface: IXHROverride = null;
-                    httpInterface = _getSenderInterface([TransportType.Xhr, TransportType.Fetch], false);
+
+                    // Prefix any user requested transport(s) values
+                    let theTransports: TransportType[] = _prependTransports([TransportType.Xhr, TransportType.Fetch], senderConfig.transports);
+
+                    if (senderConfig.isBeaconApiDisabled || !isBeaconsSupported()){
+                        // remove beacon from theTransports
+                        theTransports = theTransports.filter(transport => transport !== TransportType.Beacon);
+                    }
+
+                    httpInterface = _getSenderInterface(theTransports, false);
                   
                     let xhrInterface = { sendPOST: _xhrSender} as IXHROverride;
                     _xhrSend = (payload: string[], isAsync: boolean) => {
                         return _doSend(xhrInterface, payload, isAsync);
                     };
     
-    
-                    if (!senderConfig.isBeaconApiDisabled && isBeaconsSupported()) {
-                        // Config is set to always used beacon sending
+                    // prioritize beacon over fetch and xhr
+                    if (!senderConfig.isBeaconApiDisabled && isBeaconsSupported() && !senderConfig.transports) {
```

Comment:
Once you add the check to the `_getSenderInterface` for the `isBeaconApiDisabled` then we can probably simplify this as well. 
We should also check whether they have anything in the `transports` as well as they "could" pass an empty transports...

Maybe just move all of this to above line 365, to "prepend" the beacon if they have not set the `isBeaconApiDisabled` so that if they have provided any transports they will be added first.